src/bits/transform.h

```diff
@@ -117,9 +130,109 @@ auto transformed_with_new_return_type(InputContainer &&input, Transform &&transf
 // -------------------- transform --------------------
 template <typename Container, typename UnaryPredicate>
 void transform(Container &input, UnaryPredicate &&predicate)
+#if __cplusplus >= 202002L
+    requires UnaryPredicateOnContainerValues<UnaryPredicate, Container>
+#endif
+
 {
     std::transform(std::begin(input), std::end(input), std::begin(input),
                    std::forward<UnaryPredicate>(predicate));
 }
 
+// -------------------- transformed_if --------------------
+namespace detail {
+    // Version used for l-values or where the container type changes
+    template <typename ResultContainer, typename InputContainer, typename Transform,
+              typename UnaryPredicate>
+    ResultContainer transformed_if(InputContainer &&input, Transform &&transform,
+                                   UnaryPredicate &&unaryPredicate, std::true_type)
+    {
+        ResultContainer result;
+        detail::reserve(result, input.size());
+        auto range = read_iterator_wrapper(std::forward<InputContainer>(input));
+        auto inserter = detail::insert_wrapper(result);
+        for (auto it = range.begin; it != range.end; ++it) {
```

Comment:
Just use the range-based for loop?

Reply:
No, then I wouldn't the benefit of the insert_wrapper, namely to make it work with QMap's